### PR TITLE
Docs: Relabel Automatic Redirects as "Incoming links: Best practices and redirects" (Diátaxis)

### PR DESCRIPTION
**Pull request by benjaoming**
After reading this, I decided to expand the scope to deal more generally with the question of how to handle incoming links.

I think this would be the main theme that the reader is tackling when reading the explanation.

I added a lot of cross-references already. Potentially, more could be added.. but I got a feeling that the page would drown in cross-references at some point.

Other than that, I think this looks like a fine improvement. All the contents were already "explanation" so pretty easy to move.

Refs: #9746 

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--9896.org.readthedocs.build/en/9896/

- Developer's documentation (`dev`): https://dev--9896.org.readthedocs.build/en/9896/

### CODE REVIEW

File: docs/user/automatic-redirects.rst [hunk 1]
```diff
@@ -1,56 +1,140 @@
...
+Questionable practice 🟡
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Avoid using specific versions in links unless you really mean that users should see that exact version.
+  Versions get outdated.
+  You can use a :ref:`version warning <versions:Version warning>` to ensure the reader is aware.
+* Keep development versions hidden so people do not find them on search engines by mistake.
+* Avoid using a public ``latest`` for development versions and do not make your *default version* a development branch.
+  Publishing development branches can mean that users are reading instructions for unreleased software or draft documentation.
+* Try to get your understanding of ``latest`` and *default version* right from the beginning and ensure you don't change them later on.
```
Comment (humitos) [inline docs/user/automatic-redirects.rst:104]: This also seems to be good advice. I think I'm a little confused about this "Questionable practice" section 🤔

File: docs/user/automatic-redirects.rst [hunk 2]
```diff
@@ -1,56 +1,140 @@
...
+
+.. tip::
+
+   404 pages are okay!
+   If documentation pages have been removed or moved,
+   it can make the maintainer of the referring website aware that they need to update their link.
+   Users will be aware that the documentation still exist but has changed.
+
+   The default Read the Docs 404 page is designed to be helpful.
+   You can also design a custom 404 page, see :ref:`hosting:Custom Not Found (404) Pages`.
```
Comment (humitos) [inline docs/user/automatic-redirects.rst:114]:
```suggestion
   The default Read the Docs 404 page is designed to be helpful,
   but you can also design your own matching the style of your site, see :ref:`hosting:Custom Not Found (404) Pages`.
```

**Review by humitos — COMMENTED** [review]
I left a bunch of comments. The content is _good_; but I think there are some parts that can be clarified more to avoid confusions.

File: docs/user/automatic-redirects.rst [hunk 3]
```diff
@@ -1,56 +1,140 @@
...
-Links to rtfd.io are treated the same way as above.
+As a documentation owner,
+you most likely want users clicking on incoming links to see the latest version of the page.
+
+Good practice ✅
+~~~~~~~~~~~~~~~~
+
+* Use ``/page/path/to/page.html`` if you are linking to the page in the default version of the default language.
+* Use ``/en/latest/path/to/page.html`` if you want to be specific about the language.
+* If you move a page that likely has incoming references, :doc:`create a redirect rule </user-defined-redirects>`.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:87]: Maybe we should suggest to use Traffic Analytics to be informed before making decisions to (re)move content.

File: docs/user/automatic-redirects.rst [hunk 4]
```diff
@@ -1,56 +1,140 @@
...
+This article explains the redirects that are automatically active for all Read the Docs projects and how they are useful.
+Automatic redirects on Read the Docs are supported long-term.
 
-Redirecting to a Page
+Redirecting to a page
 ---------------------
 
 You can link to a specific page and have it redirect to your default version.
 This is done with the ``/page/`` URL prefix.
 You can reach this page by going to https://docs.readthedocs.io/page/automatic-redirects.html.
```
[the remark this answers is not in the document]
Comment (ericholscher) [reply docs/user/automatic-redirects.rst:13]: I think that just adds words without saying anything. The original seems clear?

**Review by ericholscher — COMMENTED** [review]
Manuel did a great copy edit here, and I agree with most of the feedback. I'd also give some feedback at a high level which seems like this page is trying to mix reference & explanation and it doesn't work well. The section on `permalinks` feels really odd and different in tone than the rest of the article.

I think we should probably consider moving this back to a feature reference and an explanation. I also wouldn't frame the explanation as a question, since we don't do that anywhere else? Similarly, the best practices & "what not to do" seem odd for a explanation -- though I'm not 100% sure where they belong.

File: docs/user/automatic-redirects.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (humitos) [reply docs/user/automatic-redirects.rst:13]: Maybe it's a personal thing. I usually find hard to parse "this" when there are many things involved since it depends on the context where it's written and can be ambiguos. In those cases, I tend to replace "this" with the exact thing I want to highlight in that sentence.

File: docs/user/automatic-redirects.rst [hunk 5]
```diff
@@ -1,56 +1,140 @@
...
+
+   :doc:`/user-defined-redirects`
+     If you delete or move a page,
+     you can setup a redirect in place of the old location and choose where users should be redirected.
+
+
+Root URL redirect
+-----------------
 
 A link to the root of your documentation will redirect to the *default version*,
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:34]: I like both suggestions! But they were targeting the same line so I edited your comment to have both suggestions "in one"

File: docs/user/automatic-redirects.rst [hunk 6]
```diff
@@ -1,56 +1,140 @@
...
 
     docs.readthedocs.io -> docs.readthedocs.io/en/latest/
     www.pip-installer.org -> www.pip-installer.org/en/latest/
 
-This only works for the root URL, not for internal pages. It's designed to redirect people from ``http://pip.readthedocs.io/`` to the default version of your documentation, since serving up a 404 here would be a pretty terrible user experience. (If your "develop" branch was designated as your default version, then it would redirect to ``http://pip.readthedocs.io/en/develop``.) But, it's not a universal redirecting solution. So, for example, a link to an internal page like ``http://pip.readthedocs.io/usage.html`` doesn't redirect to ``http://pip.readthedocs.io/en/latest/usage.html``.
+.. warning::
+
+   This only works for the root URL, not for internal pages.
+   It's designed to redirect people from ``http://pip.readthedocs.io/`` to the default version of your documentation,
+   since serving up a 404 here would be a pretty terrible user experience.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:48]: I think this is @ericholscher talking :) I liked it when I read it the first time, so I just left it.. but of course always safer to use neutral language. Let's figure this out some time in the style guide.. it's nice to have some kind of personal touch, but we have to know the _style_.

File: docs/user/automatic-redirects.rst [hunk 7]
```diff
@@ -1,56 +1,140 @@
...
+
+.. note::
+   If the "develop" branch was designated as the default version,
+   then ``http://pip.readthedocs.io/`` would redirect to ``http://pip.readthedocs.io/en/develop``.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:52]: Agreed, gonna fix that.

File: docs/user/automatic-redirects.rst [hunk 8]
```diff
@@ -1,56 +1,140 @@
...
+   But, it's not a universal redirecting solution.
+   So, for example, a link to an internal page like
+   ``http://pip.readthedocs.io/usage.html`` doesn't redirect to ``http://pip.readthedocs.io/en/latest/usage.html``.
+
+   The reasoning behind this is that RTD organizes the URLs for docs so that multiple translations and multiple versions of your docs can be organized logically and consistently for all projects that RTD hosts.
+   For the way that RTD views docs,
+   ``http://pip.readthedocs.io/en/latest/`` is the root directory for your default documentation in English, not ``http://pip.readthedocs.io/``.
+   Just like ``http://pip.readthedocs.io/en/develop/`` is the root for your development documentation in English.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:60]: Yes, this needs a clarification... let's make some `note::` or `tip::` about this. I'll come up with a suggestion.

File: docs/user/automatic-redirects.rst [hunk 9]
```diff
@@ -1,56 +1,140 @@
...
+you can choose which is the "default" version for RTD to display,
+which usually corresponds to the git branch of the most recent official release from your project.
+
+Are "permalinks" possible?
+--------------------------
 
-The reasoning behind this is that RTD organizes the URLs for docs so that multiple translations and multiple versions of your docs can be organized logically and consistently for all projects that RTD hosts. For the way that RTD views docs, ``http://pip.readthedocs.io/en/latest/`` is the root directory for your default documentation in English, not ``http://pip.readthedocs.io/``. Just like ``http://pip.readthedocs.io/en/develop/`` is the root for your development documentation in English.
+You might be familiar with permalinks from blogging.
+The idea is that a blog post receives a unique link as soon as it's published and that the link does not change afterward.
+Incoming sources can reference the blog post even though the blog changes structure or the post title changes.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:71]: I'm a bit reluctant to include it in our Glossary since we do not have _permalinks_ officially. I don't know if any documentation frameworks support this? I would love if they did, but I wanted to use the term so that people will understand that writing documentation and using a URL path is in some sense the same as creating a permalink.

File: docs/user/automatic-redirects.rst [hunk 10]
```diff
@@ -1,56 +1,140 @@
...
 
-Links to rtfd.io are treated the same way as above.
+As a documentation owner,
+you most likely want users clicking on incoming links to see the latest version of the page.
+
+Good practice ✅
+~~~~~~~~~~~~~~~~
+
+* Use ``/page/path/to/page.html`` if you are linking to the page in the default version of the default language.
+* Use ``/en/latest/path/to/page.html`` if you want to be specific about the language.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:86]: That will work well, yes! Good with a reference :sparkle:

File: docs/user/automatic-redirects.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:87]: I think we shouldn't write this "bombastically" - it should be possible to move a page without worrying too much. Later we make the point about 404s being useful.

File: docs/user/automatic-redirects.rst [hunk 11]
```diff
@@ -1,56 +1,140 @@
...
+  Renaming an article's title is great for the reader and great for SEO,
+  but this does not have to involve the URL.
+
+Questionable practice 🟡
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Avoid using specific versions in links unless you really mean that users should see that exact version.
+  Versions get outdated.
+  You can use a :ref:`version warning <versions:Version warning>` to ensure the reader is aware.
+* Keep development versions hidden so people do not find them on search engines by mistake.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:101]: Haha, yes, good catch, I must have been trying to write "positive" in a section that should phrase things in a negative way :D

File: docs/user/automatic-redirects.rst (hunk 1 above)
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:104]: I'll put the end of the section at the beginning :+1:

File: docs/user/automatic-redirects.rst [hunk 12]
```diff
@@ -1,56 +1,140 @@
...
+   it can make the maintainer of the referring website aware that they need to update their link.
+   Users will be aware that the documentation still exist but has changed.
+
+   The default Read the Docs 404 page is designed to be helpful.
+   You can also design a custom 404 page, see :ref:`hosting:Custom Not Found (404) Pages`.
+
+Shortlinks: rtfd.io
+-------------------
+
+Links to ``readthedocs.io`` are treated the same way as ``readthedocs.io``.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:119]: Not sure what happened here :scream: Will fix!

File: docs/user/automatic-redirects.rst [hunk 13]
```diff
@@ -1,56 +1,140 @@
...
           they should live at a prefix like `Redirecting to a Page`_
 
-The main challenge of URL routing in Read the Docs is handling redirects correctly. Both in the interest of redirecting older URLs that are now obsolete, and in the interest of handling "logical-looking" URLs (leaving out the lang_slug or version_slug shouldn't result in a 404), the following redirects are supported::
+The main challenge of URL routing in Read the Docs is handling redirects correctly.
+Both in the interest of redirecting older URLs that are now obsolete,
+and in the interest of handling "logical-looking" URLs (leaving out the lang_slug or version_slug shouldn't result in a 404),
+the following redirects are supported::
 
     /          -> /en/latest/
     /en/       -> /en/latest/
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:140]: Oh! I just added "legacy", maybe it's beyond deprecated.

[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:140]: I'd suggest to just remove the whole section then?

[the remark this answers is not in the document]
Comment (humitos) [reply docs/user/automatic-redirects.rst:139]: IMO, removing the whole section for things that don't work anymore as they say it works is 👍🏼

File: docs/user/automatic-redirects.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (humitos) [reply docs/user/automatic-redirects.rst:71]: > I'm a bit reluctant to include it in our Glossary since we do not have permalinks officially 

Describing a term in the glossary does not mean that we officially support it. It helps us to expand the meaning of a word that in the same way through out the whole documentation, to give the user more context without extending the explanation of the word in all the pages.

In the glossary, we can explain things that we don't support at all; but we mention them in our documentation.

File: docs/user/automatic-redirects.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (humitos) [reply docs/user/automatic-redirects.rst:86]: If we are adding a section about "Good recommendations", I'd say we should recommend to _always_ create a redirect if you change a URL. No matter if there were too many pageviews in the last month(s) or not. You never know where your URL is linked from 😄

File: docs/user/automatic-redirects.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:71]: Sounds good! It also aligns with an idea of being "friendly" to readers, i.e. make sure that they feel included when new terms are explained.

File: docs/user/automatic-redirects.rst [hunk 14]
```diff
@@ -1,56 +1,140 @@
...
+you most likely want users clicking on incoming links to see the latest version of the page.
+
+Good practice ✅
+~~~~~~~~~~~~~~~~
+
+* Use ``/page/path/to/page.html`` if you are linking to the page in the default version of the default language.
+* Use ``/en/latest/path/to/page.html`` if you want to be specific about the language.
+* If you move a page that likely has incoming references, :doc:`create a redirect rule </user-defined-redirects>`.
+* Links from other Sphinx projects should use :doc:`intersphinx </guides/intersphinx>`.
+* Use minimal page slugs that do not attract renaming suggestions.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:88]: Yes, that one has a :term: - just need to use it :smile:

File: docs/user/automatic-redirects.rst (hunk 1 above)
Comment (benjaoming) [reply docs/user/automatic-redirects.rst]: I moved it to best practice :)

File: docs/user/automatic-redirects.rst (hunk 11 above)
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst]: Fixed!

File: docs/user/automatic-redirects.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst]: Also fixed!

File: docs/user/automatic-redirects.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst]: I agree with @humitos that "this" can easily get confusing here. So I changed it. Managed to eliminate one word, so Hemingway would hopefully approve :)

**Comment by benjaoming** [comment]
> I'd also give some feedback at a high level which seems like this page is trying to mix reference & explanation and it doesn't work well.

After your comment, I changed a lot here, both the title and structure and intro.

I think the "reference" part of the article is really minimal, especially after we removed some incorrect old legacy section. The remaining reference isn't really a type of reference I'd expect anyone to visit a documentation page to look for. So ultimately, I think that the entire content is better shipped as an explanation with focus on how to handle external links. Through my own "fresh eyes" reading, I think this goal is near after the latest changes :+1:

File: docs/user/automatic-redirects.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (ericholscher) [reply docs/user/automatic-redirects.rst]: Heh yea -- sounds like something I would write :)

**Comment by ericholscher** [comment]
I will give it a read over tomorrow. I ran out of energy today 😴

**Comment by ericholscher** [comment]
Talked about this on our call today. I think we have good next steps:

* Combine User-defined & Automatic redirects into a `Redirects` reference
* Keep the explanation at a high-level around best practices and permalink explainer
*

**Comment by benjaoming** [comment]
Just noticed that we have this page: https://docs.readthedocs.io/en/stable/guides/deprecating-content.html

File: docs/user/guides/redirects.rst [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+  This option is often relevant when moving a project from a former host to Read the Docs.
+  In this case, often URL paths hierarchies are redirected.
+
+  Read more about this option in :ref:`user-defined-redirects:Prefix redirects`
+
+Page redirect
+  With this option,
+  you can specify a page in your documentation to redirect elsewhere.
+  The rule triggers no matter the version of your documentation that the user is visiting.
+  This rule can also redirect to another website.
```
Comment (benjaoming) [inline docs/user/guides/redirects.rst:56]: This is from the original reference. However, the preview function doesn't reflect this. Either that's a bug or the information in the reference was incorrect. Will need to check this...

Comment (benjaoming) [reply docs/user/guides/redirects.rst:56]: It checks out, so it's a bug in the UI that I already reported.

**Comment by benjaoming** [comment]
@ericholscher what you mentioned in https://github.com/readthedocs/readthedocs.org/pull/9896#issuecomment-1398839627 from our earlier call this week is implemented here and ready for review :+1:

File: docs/user/user-defined-redirects.rst [hunk 16]
```diff
@@ -1,39 +1,116 @@
...
 
-We offer a few different type of redirects based on what you want to do.
+Built-in redirects
+------------------
+
+This article explains the redirects that are automatically active for all Read the Docs projects and how they are useful.
+Automatic redirects on Read the Docs are supported long-term.
+
+Built-in redirects are especially useful for creating and sharing **incoming links**
+which is discussed indepth in :doc:`/automatic-redirects`.
```
Comment (ericholscher) [inline docs/user/user-defined-redirects.rst:51]:
```suggestion
This section explains the redirects that are automatically active for all projects and how they are useful.
Built-in redirects are especially useful for creating and sharing incoming links,
which is discussed indepth in :doc:`/automatic-redirects`.
```

File: docs/user/user-defined-redirects.rst [hunk 17]
```diff
@@ -1,39 +1,116 @@
...
+
+A link to the root of your documentation (`<slug>.readthedocs.io/`) will redirect to the  :term:`default version`,
+as set in your project settings.
+
+This works for both readthedocs.io (|org_brand|), readthedocs-hosted.com (|com_brand|), and :doc:`custom domains </custom-domains>`.
+
+For example::
+
+    docs.readthedocs.io -> docs.readthedocs.io/en/latest/
+    www.pip-installer.org -> www.pip-installer.org/en/latest/
```
Comment (ericholscher) [inline docs/user/user-defined-redirects.rst:78]:
```suggestion
    docs.readthedocs.io -> docs.readthedocs.io/en/stable/
```

File: docs/user/user-defined-redirects.rst [hunk 18]
```diff
@@ -1,39 +1,116 @@
...
+
+.. warning::
+
+   This only works for the root URL, not for internal pages.
```
Comment (ericholscher) [inline docs/user/user-defined-redirects.rst:82]: What are internal pages?

File: docs/user/user-defined-redirects.rst [hunk 19]
```diff
@@ -1,39 +1,116 @@
...
+   then ``http://pip.readthedocs.io/`` would redirect to ``http://pip.readthedocs.io/en/develop``.
+   But, it's not a universal redirecting solution.
+   So, for example, a link to an internal page like
+   ``http://pip.readthedocs.io/usage.html`` doesn't redirect to ``http://pip.readthedocs.io/en/latest/usage.html``.
+
+   The reasoning behind this is that RTD organizes the URLs for docs so that multiple translations and multiple versions of your docs can be organized logically and consistently for all projects that RTD hosts.
+   For the way that RTD views docs,
+   ``http://pip.readthedocs.io/en/latest/`` is the root directory for your default documentation in English, not ``http://pip.readthedocs.io/``.
+   Just like ``http://pip.readthedocs.io/en/develop/`` is the root for your development documentation in English.
+
```
Comment (ericholscher) [inline docs/user/user-defined-redirects.rst:97]:
```suggestion
```

Seems this both of these notes say the same thing? I don't like using examples other people control, so we should just remove it.

File: docs/user/user-defined-redirects.rst [hunk 20]
```diff
@@ -1,39 +1,116 @@
...
+which usually corresponds to the git branch of the most recent official release from your project.
+
+Shortlink with ``https://*.rtfd.io``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Links to ``rtfd.io`` are treated the same way as ``readthedocs.io``.
+They redirect the root URL to the default version of the project.
+They are intended to be easy and short for people to type.
+
+You can reach these docs at https://docs.rtfd.io.
```
Comment (ericholscher) [inline docs/user/user-defined-redirects.rst:109]:
```suggestion
You can choose which is the "default" version for Read the Docs to display.
This usually corresponds to version of the most recent official release from your project.

Shortlink with ``https://<slug>.rtfd.io``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Links to ``rtfd.io`` are treated the same way as ``readthedocs.io``.
They are intended to be easy and short for people to type.

You can reach these docs at https://docs.rtfd.io.
```

**Review by ericholscher — COMMENTED** [review]
Lots of small nitpicks, but overall this is good.

File: docs/user/automatic-redirects.rst [hunk 21]
```diff
@@ -82,63 +51,40 @@ you most likely want users clicking on incoming links to see the latest version
 Good practice ✅
 ~~~~~~~~~~~~~~~~
 
-* Use ``/page/path/to/page.html`` if you are linking to the page in the default version of the default language.
-* Use ``/en/latest/path/to/page.html`` if you want to be specific about the language.
-* If you move a page that likely has incoming references, :doc:`create a redirect rule </user-defined-redirects>`.
+* Use `page redirects <user-defined-redirects:Page redirects>`_ if you are linking to the page in the default version of the default language.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:54]: :100:

File: docs/user/automatic-redirects.rst [hunk 22]
```diff
@@ -82,63 +51,40 @@ you most likely want users clicking on incoming links to see the latest version
...
+* If you move a page that likely has incoming references, :doc:`create a custom redirect rule </guides/redirects>`.
 * Links from other Sphinx projects should use :doc:`intersphinx </guides/intersphinx>`.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:56]: good question... I guess from an author's POV it's a bit of a hard mission to check if incoming sources are using intersphinx.

Seems good to also mention "thou shalt cross-reference other Sphinx projects as thou wanteth other documentation projects to cross-reference your Sphinx project"

Must be better to not live in a glasshouse... also not throw stones actually

[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:56]: What I mean is that I think it's cool to also mention tips about how to reference other Sphinx projects in a good way :+1:

File: docs/user/automatic-redirects.rst [hunk 23]
```diff
@@ -82,63 +51,40 @@ you most likely want users clicking on incoming links to see the latest version
...
-* Use minimal page slugs that do not attract renaming suggestions.
+* Use minimal page :term:`slugs <slug>` that do not attract renaming suggestions.
 * When possible,
   keep original file names rather than going for low-impact URL renaming.
   Renaming an article's title is great for the reader and great for SEO,
   but this does not have to involve the URL.
+* Try to get your understanding of ``latest`` and *default version* right from the beginning and ensure you don't change them later on.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:62]: `latest` refers to a URL path, but it's not so clear

Will try with this:

```suggestion
* Establish your understanding of the *latest* and *default version* of your documentation at the beginning. Changing their meaning is very disruptive to incoming links.
```

File: docs/user/automatic-redirects.rst [hunk 24]
```diff
@@ -82,63 +51,40 @@ you most likely want users clicking on incoming links to see the latest version
...
+* Keep development versions hidden so people do not find them on search engines by mistake.
+  Publishing documentation at the wrong URL can be costly if the URL starts circulating.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:64]:
```suggestion
  This is the best way to ensure that nobody links to URLs that are intended for development purposes.
```

File: docs/user/automatic-redirects.rst [hunk 25]
```diff
@@ -82,63 +51,40 @@ you most likely want users clicking on incoming links to see the latest version
...
+   Using Sphinx?
+     If you are using ``:ref:`` for :doc:`cross-referencing </guides/cross-referencing-with-sphinx>`, you may add as many reference labels to a headline as you need,
+     keeping old reference labels. This will make refactoring documentation easier and avoid that external projects
+     referencing your documentation through :doc:`intersphinx </guides/intersphinx>` break.
 
 Questionable practice 🟡
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-* Avoid using specific versions in links unless you really mean that users should see that exact version.
+* Avoid using specific versions in links unless you really intend users to see that exact version.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:77]: I think the sentence is a bit more cryptic now, but it's shorter :+1:

File: docs/user/glossary.rst [hunk 26]
```diff
@@ -7,10 +7,25 @@ Glossary
       `Main page <https://readthedocs.org/dashboard>`_ where you can see all your projects with their build status
       and import a new project.
 
+   default version
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/glossary.rst:10]: This is a mistake. Will add references.

File: docs/user/glossary.rst [hunk 27]
```diff
@@ -7,10 +7,25 @@ Glossary
...
+      A documentation has a *default version*, usually understood to be the latest stable version of a project.
+      Most documentation projects will contain both older versions of documentation and the latest development version.
+      The *default version* can be fully configured to point to a specific Git branch or any other released version.
+      In the most common scenario,
+      the default version points to the ``main`` branch of a Git repository since it usually points to the latest stable release.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/glossary.rst:15]: That's a much clearer way to explain it :+1: Also nice with the URL-perspective since it was intended to be included in pages about URLs and links :)

File: docs/user/user-defined-redirects.rst (hunk 18 above)
Comment (benjaoming) [reply docs/user/user-defined-redirects.rst:82]: I don't know + this whole warning box is weird. Look at the warning text below.

Gonna delete it...

Comment (benjaoming) [reply docs/user/user-defined-redirects.rst:82]: I think we should combine the two warnings to a single `note::` that says

> You cannot use the root redirect to reference specific pages. `/` redirects to the default version, but `/some/page.html` will not redirect to `/en/latest/some/page.html`. Use the `/page/` redirect rule for this purpose.

File: docs/user/user-defined-redirects.rst (hunk 20 above)
Comment (benjaoming) [reply docs/user/user-defined-redirects.rst:109]: Thanks! Modified your suggestion a tiny bit :+1:

File: docs/user/automatic-redirects.rst [hunk 28]
```diff
@@ -1,72 +1,41 @@
...
-
-Among all the multiple versions of docs,
-you can choose which is the "default" version for RTD to display,
-which usually corresponds to the git branch of the most recent official release from your project.
-
-Are "permalinks" possible?
---------------------------
-
-You might be familiar with permalinks from blogging.
+You might be familiar with :term:`permalinks <permalink>` from blogging.
```
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst:38]: it's a very good article that I used as inspiration for the glossary entry :)

but yeah, I'll trim that entry since it gives the impression that it's a term that RTD has a special definition of.

File: docs/user/user-defined-redirects.rst (hunk 18 above)
Comment (benjaoming) [reply docs/user/user-defined-redirects.rst:79]: Made it a warning box.

**Comment by benjaoming** [comment]
@ericholscher PR is ready for another review :eyes:

File: docs/user/automatic-redirects.rst (hunk 25 above)
[the remark this answers is not in the document]
Comment (ericholscher) [reply docs/user/automatic-redirects.rst]: Not sure I'd say it's more cryptic, and not sure if this response is sarcasm?

**Review by ericholscher — APPROVED** [review]
Changes look good 👍

File: docs/user/automatic-redirects.rst (hunk 25 above)
[the remark this answers is not in the document]
Comment (benjaoming) [reply docs/user/automatic-redirects.rst]: No, it wasn't. When text is trimmed out, *some* meaning can be lost. In this case, it's about clarifying and spelling out *unless you really intend users to see that exact version*. Because the context is a list of "Questionable practice 🟡", then the idea of spelling out why the practice can be questionable is a bit more in the spotlight :)

When shortening to *unless users need that exact version*, there's the risk that someone doesn't get the meaning."Yes, they should see that version". The overly long sentence was supposed to capture the afterthought.

Maybe not successful, I get that :+1:
